katello/hooks/pre_commit/09-version_locking.rb

```diff
@@ -0,0 +1,21 @@
+# evaluate version locking settings
+cli_param = app_value(:lock_package_versions)
+custom_config_value = get_custom_config(:lock_package_versions)
+lock_versions = cli_param.nil? ? !!custom_config_value : cli_param
+
+if lock_versions != custom_config_value
+  store_custom_config(:lock_package_versions, lock_versions)
+  kafo.config.configure_application
+end
+
+# unlock packages if locked
+`command -v foreman-maintain`
+if $?.success?
+  `foreman-maintain packages is-locked --assumeyes`
+  if $?.exitstatus == 0
+    Kafo::Helpers.log_and_say :info, "Package versions are locked. Continuing with unlock."
+    Kafo::Helpers.execute('foreman-maintain packages unlock --assumeyes')
```

Comment:
I assume unlocking is idempotent so do we even need to check if it's locked? IIRC the execute helper also prints + logs the output so the log line might not be needed. This will mostly depend on the output of foreman-maintain and I'll leave this up to you.

Reply:
@ekohl the unlock will probably be expensive on Debian so I wanted to do it only when necessary. On yum based systems it is probably not the case and the check could be skipped. I'll keep the foreman-maintain output in log only.